clients/proxy_pool_registry.go

```diff
@@ -0,0 +1,119 @@
+package clients
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/erpc/erpc/common"
+	"github.com/rs/zerolog"
+)
+
+// contains a set of http.Clients (each configured with a different proxy).
+type ProxyPool struct {
+	ID      string
+	clients []*http.Client
+	counter uint64 // Add atomic counter for round-robin
+}
+
+// returns a round-robin client from the pool.
+func (p *ProxyPool) GetClient() *http.Client {
+	if len(p.clients) == 0 {
+		// Fallback: if for some reason the pool has zero clients
+		return &http.Client{Timeout: 60 * time.Second}
```

Comment:
This path shouldn't occur I suggest to "panic()" because if it for some reason happens, it'll be hidden + creating too many Client objects in memory, and causing OOM or high CPU usage.